### ISSUE TITLE
test_package settings not overriding conan.conf

### ISSUE THREAD

**Issue by tamboril** [conan-io/conan#1298]
I had 

      compiler=gcc
      compiler.version=6.3

in my `conan.conf` and tried to run `conan test_package -scompiler=gcc -scompiler.version=7.1`.
It kept erroring out as if the `compiler.version` were `6.3`.  The command-line setting did not override
that in `conan.conf`.  Once I edited `conan.conf` to change the `compiler.version` to `7.1`, the `test_package` worked correctly.

**Comment by memsharded** [comment]
Hi @tamboril 

I am running a test with your configuration and not able to reproduce it:

```python
    def test_settings_test(self):
        client = TestClient()
        conanfile = '''
from conans import ConanFile

class HelloConan(ConanFile):
    name = "Hello"
    version = "0.1"
    settings = "os", "compiler"
    def build(self):
        self.output.info("Hello OS: %s" % self.settings.os)
        self.output.info("Hello compiler: %s" % self.settings.compiler)
        self.output.info("Hello compiler.version: %s" % self.settings.compiler.version)
        '''
        test_package = '''
import os
from conans import ConanFile

class HelloTestConan(ConanFile):
    name = "HelloTest"
    version = "0.1"
    requires = "Hello/0.1@lasote/testing"
    settings = "os", "compiler"

    def build(self):
        self.output.info("Test OS: %s" % self.settings.os)
        self.output.info("Test compiler: %s" % self.settings.compiler)
        self.output.info("Test compiler.version: %s" % self.settings.compiler.version)

    def test(self):
        self.output.info("Test test() OS: %s" % self.settings.os)
        self.output.info("Test test() compiler: %s" % self.settings.compiler)
        self.output.info("Test test()  compiler.version: %s" % self.settings.compiler.version)
'''
        text = load(client.client_cache.conan_conf_path)
        text = text[:text.find("[settings_defaults]")]
        text += """[settings_defaults]
compiler = gcc
compiler.version = 6.1
compiler.libcxx = libstdc++11
"""
        save(client.client_cache.conan_conf_path, text)
        client.save({"conanfile.py": conanfile, "test_package/conanfile.py": test_package})
        client.run("test_package -s os=Linux -s compiler=gcc -s compiler.version=4.9 -s compiler.libcxx=libstdc++")
        print client.user_io.out
```

The output seems to be correct:
```bash
Hello/0.1@lasote/testing: Hello OS: Linux
Hello/0.1@lasote/testing: Hello compiler: gcc
Hello/0.1@lasote/testing: Hello compiler.version: 4.9
...
Project: Test OS: Linux
Project: Test compiler: gcc
Project: Test compiler.version: 4.9
Project: Test test() OS: Linux
Project: Test test() compiler: gcc
Project: Test test()  compiler.version: 4.9
```

Could you please provide more details on the error, so I can re-check? Thanks!

**Comment by tamboril** [comment]
I made a simplified example in the attached zip.  I also included conf and yml files.
[memsharded.zip](https://github.com/conan-io/conan/files/1018986/memsharded.zip)

The problem is this:
```
conan export jharris/testing
conan test_package -scompiler=gcc -scompiler.version=7.1
ERROR: /home/jharris/repos/conan-recipes/dvconan/memsharded/test_package/conanfile.py: Error while initializing settings. Invalid setting '6.3' is not a valid 'settings.compiler.version' value.
Possible values are ['7.1']
```

If I edit `~/.conan/conan.conf` (a copy of which I put into the ZIP), to set `compiler.version` to `7.1`, then it works.

**Comment by memsharded** [comment]
Ok, thanks, very useful. The issue comes because of the ``"compiler": {"gcc": {"version": ["7.1"]}}`` constraint in the recipe. Not sure how to quick fix it, lets have a look for 0.24

**Comment by memsharded** [comment]
I have added a test to make sure this works now: https://github.com/conan-io/conan/pull/1461

This problem was fixed by recent work moving conan.conf ``[default_settings]`` to a default_profile: https://github.com/conan-io/conan/pull/1446

**Comment by memsharded** [comment]
Released in 0.25
